lib/src/test/java/com/cloudbees/groovy/cps/BasicTest.java

```diff
@@ -0,0 +1,324 @@
+package com.cloudbees.groovy.cps;
+
+import com.cloudbees.groovy.cps.impl.CpsCallableInvocation;
+import com.cloudbees.groovy.cps.impl.CpsFunction;
+import org.codehaus.groovy.runtime.ScriptBytecodeAdapter;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+
+import static java.util.Arrays.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * @author Kohsuke Kawaguchi
+ */
+public class BasicTest {
+    Builder b = new Builder(MethodLocation.UNKNOWN);
+
+    // useful fragment of expressions
+    Block $x = b.localVariable("x");
+    Block $y = b.localVariable("y");
+    Block $z = b.localVariable("z");
+
+    /**
+     * Evaluates the given body and return the yielded value.
+     */
+    private <T> T run(Block... bodies) {
+        try {
+            Env e = Envs.empty();
+            Next p = new Next(b.block(bodies), e, Continuation.HALT);
+            return (T) p.run().yield.wrapReplay();
+        } catch (InvocationTargetException x) {
+            throw new AssertionError(x);
+        }
+    }
+
+
+    // 3    => 3
+    @Test
+    public void constant() {
+        assertThat(run(b.constant(3)), equalTo(3));
```

Comment:
Updating to Java 8 caused this class to fail to compile due to ambiguity between `assertEquals(Object, Object)` and `assertEquals($primitive, $primitive)` in various tests. To fix it, I switched to using `assertThat` and `equalTo`, which should be more or less the same as using the `(Object, Object)` overloads of `assertEquals`.